plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -454,7 +454,8 @@ public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
         ImmutableMap.Builder<JdbcColumnHandle, JdbcColumnHandle> newLeftColumnsBuilder = ImmutableMap.builder();
         for (JdbcColumnHandle column : jdbcClient.getColumns(session, leftHandle)) {
             newLeftColumnsBuilder.put(column, JdbcColumnHandle.builderFrom(column)
-                    .setColumnName(column.getColumnName() + "_" + nextSyntheticColumnId)
+                    .setColumnName(column.getOriginalColumnName() + "_" + session.getAlwaysIncreasingNumber())
```

Comment:
I wasn't sure whether we can re-use the nextSyntheticColumnId, as it seems to be using the max() function. If anyone knows whether 'nextSyntheticColumnId' has the property that it always increases monothonicaly, we could reuse it. However, keep in mind that the same change would need to be applied for 'aggregation' (here the change is only for 'join'), so I'd be better to come up with a generic solution.